src/garage/torch/distributions/tanh_normal.py

```diff
@@ -0,0 +1,305 @@
+
+        The result will enumerate over dimension 0, so the shape
+        of the result will be `(cardinality,) + batch_shape + event_shape`
+        (where `event_shape = ()` for univariate distributions).
+
+        Note that this enumerates over all batched tensors in lock-step
+        `[[0, 0], [1, 1], ...]`. With `expand=False`, enumeration happens
+        along dim 0, but with the remaining batch dimensions being
+        singleton dimensions, `[[0], [1], ..`.
+
+        To iterate over the full Cartesian product use
+        `itertools.product(m.enumerate_support())`.
+
+        Args:
+            expand (bool): whether to expand the support over the
+                batch dims to match the distribution's `batch_shape`.
+
+        Note:
+            Calls the enumerate_support function of the underlying normal
+            distribution.
+
+        Returns:
+            torch.Tensor: Tensor iterating over dimension 0.
+
+        """
+        return self._normal.enumerate_support(expand)
+
+    @property
+    def mean(self):
+        """Returns mean of the distribution.
+
+        Note: This is the mean of the underlying normal distribution
+              with the Tanh transformation applied.
+
+        Returns:
+            torch.Tensor: mean of the distribution
+
+        """
+        return torch.tanh(self._normal.mean)
+
+    @property
+    def variance(self):
+        """Returns variance of the underlying normal distribution.
+
+        Returns:
+            torch.Tensor: variance of the underlying normal distribution.
+
+        """
+        return self._normal.variance
+
+    def entropy(self):
+        """Returns entropy of the underlying normal distribution.
+
+        Returns:
+            torch.Tensor: entropy of the underlying normal distribution.
+
+        """
+        return self._normal.entropy()
+
+    @staticmethod
+    def _clip_but_pass_gradient(x, lower=0., upper=1.):
+        """Clipping function that allows for gradients to flow through.
+
+        Args:
+            x(torch.tensor): value to be clipped
+            lower(float): lower bound of clipping
+            upper(float): upper bound of clipping
+
+        Returns:
+            torch.Tensor: x clipped between lower and upper.
+
+        """
+        clip_up = (x > upper).float()
+        clip_low = (x < lower).float()
+        with torch.no_grad():
+            clip = ((upper - x) * clip_up + (lower - x) * clip_low)
+        return x + clip
+
+    def __str__(self):
+        """Returns the Name of the class.
+
+        Returns:
+            str: The class name.
+
+        """
+        return 'TanhNormal'
+
+    def __repr__(self):
+        """Returns the parameterization of the distribution.
+
+        Returns:
+            str: The parameterization of the distribution and underlying
+                distribution.
+
+        """
+        return ('TanhNormal Dist: Mean {}, stdev {}\n \
```

Comment:
probably better to keep this in a single line, and to do something succinct like

`TanhNormal<Normal(loc={}, scale={})>`

if torch.distributions already has conventions for `__str__` and `__repr__`, follow how they do it.